cli/src/commands/jobs.rs

```diff
@@ -3,66 +3,36 @@ use std::str::FromStr;
 
 use anyhow::{anyhow, Context, Result};
 use console::style;
-use log::LevelFilter;
 use phylum_project::LockfileConfig;
 use phylum_types::types::common::{JobId, ProjectId};
-use phylum_types::types::job::{Action, JobStatusResponse};
 use phylum_types::types::package::{PackageDescriptor, PackageType};
 use reqwest::StatusCode;
 
-use crate::api::{PhylumApi, PhylumApiError};
-use crate::commands::{parse, CommandResult, CommandValue, ExitCode};
-use crate::filter::{Filter, FilterIssues};
+use crate::api::PhylumApi;
+use crate::commands::{parse, CommandResult, ExitCode};
 use crate::format::Format;
 use crate::{config, print_user_success, print_user_warning};
 
-fn handle_status<T>(
-    resp: Result<JobStatusResponse<T>, PhylumApiError>,
-    pretty: bool,
-) -> Result<Action>
-where
-    JobStatusResponse<T>: Format,
-{
-    let resp = match resp {
-        Ok(resp) => resp,
+/// Output analysis job results.
+pub async fn print_job_status(api: &mut PhylumApi, job_id: &JobId, pretty: bool) -> CommandResult {
+    let response = api.get_job_status(job_id, []).await;
+
+    // Provide nicer messages for specific errors.
+    let status = match response {
+        Ok(status) => status,
         Err(err) if err.status() == Some(StatusCode::NOT_FOUND) => {
-            print_user_warning!(
-                "No results found. Submit a lockfile for processing:\n\n\t{}\n",
-                style("phylum analyze <lock_file>").blue()
-            );
-            return Ok(Action::None);
+            print_user_warning!("No results found for JobId {job_id}.");
+            return Ok(ExitCode::Ok);
```

Comment:
You've just kept the old behavior here. But why does a 404 response code exit successfully? Seems like that should be an error.